analyzers/tests/SonarAnalyzer.UnitTest/Helpers/FilesToAnalyzeRetrieverTest.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+* SonarAnalyzer for .NET
+* Copyright (C) 2015-2021 SonarSource SA
+* mailto: contact AT sonarsource DOT com
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this program; if not, write to the Free Software Foundation,
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.UnitTest.Helpers
+{
+    [TestClass]
+    public class FilesToAnalyzeRetrieverTest
+    {
+        [TestMethod]
+        public void NonExistingFile_RetrieveFilesToAnalyze_ReturnsEmptyEnumerable()
+        {
+            var sut = new FilesToAnalyzeRetriever();
+
+            sut.RetrieveFilesToAnalyze(@"ResourceTests\FilesToAnalyze\NonExistingFile.txt").Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void EmptyFile_RetrieveFilesToAnalyze_ReturnsEmptyEnumerable()
+        {
+            var sut = new FilesToAnalyzeRetriever();
+
+            sut.RetrieveFilesToAnalyze(@"ResourceTests\FilesToAnalyze\EmptyFilesToAnalyze.txt").Should().BeEmpty();
+        }
+
+        [DataTestMethod]
+        [DataRow("")]
```

Comment:
maybe also add a non-empty string

The `null` and empty can be filtered with `IsNullOrEmpty`, however a non-empty string may cause an I/O exception ("file not found")